### PR TITLE
Remove deprecated create-tenant cli command from integration tests

### PR DESCRIPTION
**Pull request by alfrunes**
changelog: title

Signed-off-by: Alf-Rune Siqveland <alf.rune@northern.tech>

### CODE REVIEW

File: backend-tests/tests/test_deployments.py [hunk 1]
```diff
@@ -137,9 +137,9 @@ def setup_deployments_enterprise_test(clean_mongo,
     Creates two tenants, with one user each, where each user has three deployments,
     and a hundred devices each.
     """
-    tenant1 = create_tenant_test_setup('bugs-bunny', 'acme')
+    tenant1 = create_tenant_test_setup('bugs@bunny.org', 'acme')
     # Add a second tenant to make sure that the functionality does not interfere with other tenants
-    tenant2 = create_tenant_test_setup('road-runner', 'indiedev')
+    tenant2 = create_tenant_test_setup('road@runne.orgr', 'indiedev')
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_deployments.py]: orgr? I know it doesn't matter but r it's here by mistake right?

File: backend-tests/tests/test_tenantadm.py [hunk 2]
```diff
@@ -6,41 +6,46 @@
...
+                                             username="user@example.com",
+                                             password="password")
+        self.logger.debug("Tenant id: %s" % tenant_id)
 
-        # Retry login for 3 min
-        for i in range(60 * 3):
-            time.sleep(1)
+        # Retry login every 250ms for 3 min
+        for i in range(240 * 3):
+            time.sleep(0.25)
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_tenantadm.py]: do you think it makes sense to retries every 250ms?

File: backend-tests/tests/test_tenantadm.py [hunk 3]
```diff
@@ -73,23 +81,17 @@ def test_duplicate_username(self, clean_mongo_tenant_migration):
 
         try:
             self.logger.debug("Second tenant creation call")
-            out = create_org("barCorp", "user@example.com", "321password")
-            self.logger.debug("stdout: %s" % out)
+            tenant_id = tenantadm_cli.create_org(name="barCorp",
+                                                 username="user@example.com",
+                                                 password="321password")
+            self.logger.debug(
+                "Tenant id: %s (THIS SHOULD NOT BE LOGGED)" % tenant_id)
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_tenantadm.py]: ?

File: backend-tests/tests/test_tenantadm.py [hunk 4]
```diff
@@ -98,14 +100,16 @@ def test_duplicate_organization(self, clean_mongo_tenant_migration):
...
+                                             username="foo@corp.org",
+                                             password="321password")
+        self.logger.debug("Tenant id: %s" % tenant_id)
 
-        # Retry login for 3 min
-        for i in range(60 * 3):
-            time.sleep(1)
+        # Retry login every 250ms for 3 min
+        for i in range(240 * 3):
+            time.sleep(0.25)
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_tenantadm.py]: same question as above

File: backend-tests/tests/test_tenantadm.py [hunk 5]
```diff
@@ -114,13 +118,14 @@ def test_duplicate_organization(self, clean_mongo_tenant_migration):
...
+                                             username="foo@acme.com",
+                                             password="password123")
+        self.logger.debug("Tenant id: %s" % tenant_id)
 
-        # Retry login for 3 min
-        for i in range(60 * 3):
-            time.sleep(1)
+        # Retry login every 250ms for 3 min
+        for i in range(240 * 3):
+            time.sleep(0.25)
```
Comment (kjaskiewiczz) [inline backend-tests/tests/test_tenantadm.py]: same

File: tests/MenderAPI/authentication.py [hunk 6]
```diff
@@ -24,99 +26,135 @@
...
-            r = self._do_login(self.email, self.password)
+                    self.create_user(self.username, self.password)
+
+            # It might take some time for create_org to propagate the new user.
+            # Retry login for a minute.
+            for i in range(600):
+                r = self._do_login(self.username, self.password)
+                if r.status_code == 200:
+                    break
+                time.sleep(0.1)
```
Comment (kjaskiewiczz) [inline tests/MenderAPI/authentication.py]: this time you're retrying every 100ms; why not wait a second (literally)?

File: testutils/common.py [hunk 7]
```diff
@@ -141,24 +112,47 @@ def create_authset(dauthd1, dauthm, id_data, pubkey, privkey, utoken, tenant_tok
...
+    # - there usually is a slight delay in user creation
+    for i in range(600):
+        rsp = api.call(
+            "POST",
+            useradm.URL_LOGIN,
+            auth=(username, password)
+        )
+        if rsp.status_code == 200:
+            break
+        time.sleep(0.1)
```
Comment (kjaskiewiczz) [inline testutils/common.py]: same question

File: backend-tests/tests/test_tenantadm.py (hunk 3 above)
Comment (alfrunes) [reply backend-tests/tests/test_tenantadm.py]: I should probably replace this line with the `pytest.fail` statement. The above line should raise an exception.

**Comment by alfrunes** [comment]
@kacf @kjaskiewiczz; Could you take a quick look and see if everything looks good?

**Comment by kacf** [comment]
There are a lot of unrelated changes here, including a lot of whitespace changes. Can you keep those separate please? The various fixture changes also look unrelated to me, though I didn't look very closely yet.

**Comment by alfrunes** [comment]
> There are a lot of unrelated changes here, including a lot of whitespace changes. Can you keep those separate please? The various fixture changes also look unrelated to me, though I didn't look very closely yet.

Some changes are probably due to my recently added `autopep8` hook. There are a couple of unrelated changes as well, I'll see if I can stash some of the changes away from this PR. Regarding the fixtures you'll have to be more specific - I did a small change in the `create_org` function in `testutils/common.py` which snowballed into some seemingly redundant changes in some fixtures.

**Comment by alfrunes** [comment]
@kacf; I removed the formatting changes from the commit and adjusted the commit message as requested.

**Comment by alfrunes** [comment]
Tests passed on previous run - only Travis that failed to report status. Merging.

**Pull request merged**